xtable-core/src/main/java/org/apache/xtable/catalog/glue/GlueCatalogSyncClient.java

```diff
@@ -0,0 +1,234 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+ 
+package org.apache.xtable.catalog.glue;
+
+import java.time.ZonedDateTime;
+
+import lombok.Getter;
+import lombok.extern.log4j.Log4j2;
+
+import org.apache.hadoop.conf.Configuration;
+
+import com.google.common.annotations.VisibleForTesting;
+
+import org.apache.xtable.conversion.TargetCatalog;
+import org.apache.xtable.exception.CatalogSyncException;
+import org.apache.xtable.exception.NotSupportedException;
+import org.apache.xtable.model.InternalTable;
+import org.apache.xtable.model.catalog.CatalogTableIdentifier;
+import org.apache.xtable.model.storage.TableFormat;
+import org.apache.xtable.spi.sync.CatalogSyncClient;
+
+import software.amazon.awssdk.services.glue.GlueClient;
+import software.amazon.awssdk.services.glue.model.CreateDatabaseRequest;
+import software.amazon.awssdk.services.glue.model.DatabaseInput;
+import software.amazon.awssdk.services.glue.model.DeleteTableRequest;
+import software.amazon.awssdk.services.glue.model.EntityNotFoundException;
+import software.amazon.awssdk.services.glue.model.GetDatabaseRequest;
+import software.amazon.awssdk.services.glue.model.GetTableRequest;
+import software.amazon.awssdk.services.glue.model.GetTableResponse;
+import software.amazon.awssdk.services.glue.model.Table;
+
+@Log4j2
+public class GlueCatalogSyncClient implements CatalogSyncClient<Table> {
+
+  protected static final String GLUE_EXTERNAL_TABLE_TYPE = "EXTERNAL_TABLE";
+  private static final String TEMP_SUFFIX = "_temp";
+
+  protected final TargetCatalog targetCatalog;
+  @Getter protected final GlueClient glueClient;
+  @Getter protected final GlueCatalogConfig glueCatalogConfig;
+  @Getter protected final Configuration configuration;
+  @Getter protected final GlueSchemaExtractor schemaExtractor;
+  protected final IcebergGlueCatalogSyncOperations icebergGlueCatalogSyncOperations;
```

Comment:
@vinishjail97 Is this fine or you want some sort of manager / factory here? For factory, we might need to pass table format as targetCatalog does not have table format

Reply:
`IcebergGlueCatalogSyncOperations` shouldn't create the table IMO, it should return the requests or the glue table objects for a given `InternalTable` object. 

The glueClient is not initialized differently for iceberg, hudi or delta but the requests being passed to it are different so these classes for each format should be focussed on generating those requests. Right now the relationship is still like a extending a base class.

Reply:
Makes sense, updated